### PR TITLE
Add dotnet-format

### LINKED ISSUE #39: dotnet-format
https://github.com/dotnet/format

### PR DESCRIPTION
**Pull request by j2ghz**
Closes #39

### CODE REVIEW

**Comment by j2ghz** [comment]
I don't know how to make the test pass. The tool can only work on C# projects, not individual files. Even if a list of files is passed, the tool requires them to be part of a C# project.

More info: https://github.com/dotnet/format#how-to-use

**Comment by pbrisbin** [comment]
> The tool can only work on C# projects, not individual files.

Can it only run on every single file in the project, or is it just that it has to _be in_ a project to function, but it can still be made to only operate on the specific files passed?

If it's the latter, we should be able to make that work (see below), but if it's the former then this won't work (as is) as a Restyler, unfortunately.

> I don't know how to make the test pass. ... the tool requires them to be part of a C# project.

Does "be part of a C# project" just mean that some manifest or details file exists? If so, the JDT Restyler is similar. I requires a `pom.xml` exists to describe the (Java) project structure. Note that the file just has to exist, it still Restyles the paths given on the command-line. If you take a look at [the JDT tests](https://github.com/restyled-io/restylers/blob/master/jdt/info.yaml#L29-L41), it uses the `support` key to write one, you would have to do something like that.

**Comment by j2ghz** [comment]
> Can it only run on every single file in the project, ...

It's possible to use `--files` to specify which files to format.


> Does "be part of a C# project" just mean that some manifest or details file exists?

Yes,  `*.csproj`. I'll try with the `support` key.

File: dotnet-format/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM mcr.microsoft.com/dotnet/core/sdk
+RUN dotnet tool install -g dotnet-format
+ENV PATH="/root/.dotnet/tools:${PATH}"
+RUN mkdir -p /code
+WORKDIR /code
+#COPY files /
+CMD ["dotnet", "format", "--help"]
```
Comment (pbrisbin) [inline dotnet-format/Dockerfile]: :art: missing final newline.

File: dotnet-format/info.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+image: restyled/restyler-dotnet-format:v0.0.1
+command:
+  - dotnet-format
+  - --files
+arguments: []
+include:
+  - "**/*.cs"
+  - "**/*.vb"
+supports_arg_sep: false
+# It is supported, but only comma separated
```
Comment (pbrisbin) [inline dotnet-format/info.yaml]: Could add a shim-script to accept path arguments and build the comma-separated invocation.

File: dotnet-format/Dockerfile (hunk 1 above)
Comment (pbrisbin) [inline dotnet-format/Dockerfile]: It's `dotnet format` here, but `dotnet-format` in the info `command` :thinking:

Comment (j2ghz) [reply dotnet-format/Dockerfile]: I'm not sure which is better. `dotnet-format` is the executable, but if you call `dotnet format`, `dotnet` will call `dotnet-format` and that's how it's documented here: https://github.com/dotnet/format#how-to-use

**Pull request merged**